File: ContinuousIntegration-AOLV/Java/src/main/java/domain/Manager.java
package domain;

public class Manager extends Employee {
	
	public Manager(String firstName, String lastName, float fixedSalary)
	{
		super(firstName, lastName, fixedSalary);
	}

	/**
	 * @return the salary with additions and deductions
	 */
	public float calculateSalaryAfterAdditionsAndDeductions() {
		float addicionalBenefits = calculateAddicionalBenefits();
		float pensionFounds = this.fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return addicionalBenefits + fixedSalary - pensionFounds - tax;
	}

	private float calculateAddicionalBenefits() {
		return this.subordinates.size() * 20;
	}
}

File: Net/OnlineStore.Domain/Manager.cs
namespace OnlineStore.Domain
{
    public class Manager : Employee
    {
        public Manager(string firstName, string lastName, decimal fixedSalary)
            : base(firstName, lastName, fixedSalary)
        {
        }
        
        public decimal SalaryAfterAdditionsAndDeductions()
        {
            decimal benefits = this.SalaryBenefits();
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (this.FixedSalary > 3500)
                tax = this.FixedSalary * 5 / 100;
            return this.FixedSalary + benefits - pensionFounds - tax;
        }

        private decimal SalaryBenefits()
        {
            return this.subordinates.Count * 20;
        }
    }
}